### PR TITLE
Handle enabled currencies and check trade time on creation

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot_trading/constants.py [hunk 1]
```diff
@@ -99,6 +99,9 @@
 BALANCE_PROFITABILITY_CHANNEL = "BalanceProfitability"
 POSITIONS_CHANNEL = "Positions"
 
+# 946742400 is 01/01/2020, if trade time is lower, there is an issue.
+MINIMUM_VAL_TRADE_TIME = 946742400
```
Comment (Herklos) [inline octobot_trading/constants.py:103]: If we load user exchange history, couldn't there be a trade in 2019 ?

File: octobot_trading/exchanges/exchange_config_data.py [hunk 2]
```diff
@@ -29,9 +29,15 @@ def __init__(self, exchange_manager):
         self.exchange_manager = exchange_manager
         self.config = exchange_manager.config
 
+        # dict of exchange supported pairs by enabled currencies from self.config
         self.traded_cryptocurrencies = {}
+        # list of exchange supported enabled pairs from self.config
         self.traded_symbol_pairs = []
+        # list of exchange supported pairs from self.config
+        self.existing_symbol_pairs = []
```
Comment (Herklos) [inline octobot_trading/exchanges/exchange_config_data.py:37]: I would have prefer something like 
```suggestion
        self.config_symbol_pairs = []
```

File: octobot_trading/exchanges/exchange_config_data.py [hunk 3]
```diff
@@ -65,38 +66,54 @@ async def handle_time_frame_update(self, exchange: str, exchange_id: str, time_f
 
     def _set_config_traded_pairs(self):
```
Comment (Herklos) [inline octobot_trading/exchanges/exchange_config_data.py:67]: This method got really big

File: octobot_trading/exchanges/exchanges.py [hunk 4]
```diff
@@ -25,6 +25,9 @@ def __init__(self, exchange_manager, matrix_id):
         self.exchange_name = exchange_manager.exchange_name
         self.id = exchange_manager.id
         self.matrix_id = matrix_id
+        # use only enabled currencies
```
Comment (Herklos) [inline octobot_trading/exchanges/exchanges.py:28]: :+1:

File: octobot_trading/personal_data/portfolios/portfolio_profitability.py [hunk 5]
```diff
@@ -133,10 +132,9 @@ def _only_symbol_currency_filter(self, currency_dict):
     def _init_traded_currencies_without_market_specific(self):
         """
         Initialize traded currencies without market specific set
-        TODO do not use config[CONFIG_CRYPTO_CURRENCIES]
+        Use exchange_config.existing_symbol_pairs to take every config pair into account including disabled ones
```
Comment (Herklos) [inline octobot_trading/personal_data/portfolios/portfolio_profitability.py:135]: :+1:

File: octobot_trading/personal_data/trades/trade_factory.py [hunk 6]
```diff
@@ -44,6 +46,9 @@ def create_trade_from_order(order,
                             canceled_time=canceled_time,
                             creation_time=creation_time,
                             executed_time=executed_time)
+    if trade.get_time() < constants.MINIMUM_VAL_TRADE_TIME:
```
Comment (Herklos) [inline octobot_trading/personal_data/trades/trade_factory.py:49]: Same as above about MINIMUM_VAL_TRADE_TIME

File: octobot_trading/util/config_util.pxd [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+
+
+""" Order class will represent an open order in the specified exchange
```
Comment (Herklos) [inline octobot_trading/util/config_util.pxd:19]: I don't think this is about order class :smile:

File: tests/util/test_config_util.py [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+def test_is_trader_enabled(config):
+    _test_enabled(config, util.is_trader_enabled, trading_constants.CONFIG_TRADER, False)
+
+
+def test_is_trader_simulator_enabled(config):
+    _test_enabled(config, util.is_trader_simulator_enabled, trading_constants.CONFIG_SIMULATOR, True)
+
+
+def test_is_currency_enabled(config):
```
Comment (Herklos) [inline tests/util/test_config_util.py:33]: :+1:

File: octobot_trading/constants.py (hunk 1 above)
Comment (GuillaumeDSM) [reply octobot_trading/constants.py:103]: oh ! comment is wrong, its 01/01/2000

Comment (Herklos) [reply octobot_trading/constants.py:103]: Oh ok :smile: that's ok then :wink:

File: octobot_trading/exchanges/exchange_config_data.py (hunk 3 above)
Comment (GuillaumeDSM) [reply octobot_trading/exchanges/exchange_config_data.py:67]: right, i'll split it

**Review by Herklos — APPROVED** [review]
:+1:

**Review by Herklos — APPROVED** [review]

**Pull request merged**